src/Functions/if.cpp

```diff
@@ -269,13 +271,22 @@ class FunctionIf : public FunctionIfBase
     static constexpr auto name = "if";
     static FunctionPtr create(ContextPtr context)
     {
-        return std::make_shared<FunctionIf>(context->getSettingsRef()[Setting::allow_experimental_variant_type] && context->getSettingsRef()[Setting::use_variant_as_common_type]);
+        return std::make_shared<FunctionIf>(
+            context->getSettingsRef()[Setting::allow_experimental_variant_type]
+                && context->getSettingsRef()[Setting::use_variant_as_common_type],
+            !context->getSettingsRef()[Setting::optimize_if_transform_strings_to_enum]);
```

Comment:
Alternatively I could change `src/Analyzer/Passes/IfTransformStringsToEnumPass.cpp` and wrap in `toLowCardinality(toString(...))` rather than just `toString(...)`. Let me know what is preferred.